<?xml version="1.0"?>
<launch>
	<arg name="prefix" default=""/>
	<arg name="node_name" default="axis_camera"/>
	<arg name="ip_address" default="192.168.0.185"/>
	<arg name="username" default="root"/>
	<arg name="password" default="R0b0tn1K"/>
    <arg name="camera_model" default="axis_m5525"/>
	<arg name="ptz_enabled" default="true"/>
	<arg name="ptz_config_file" default="$(find axis_camera)/config/$(arg camera_model)_ptz_config.yaml"/>
	<!-- enables the authentication -->
	<arg name="enable_auth" default="false"/> 

	<arg name="pan_joint" default="$(arg prefix)$(arg node_name)_pan_joint"/>
	<arg name="tilt_joint" default="$(arg prefix)$(arg node_name)_tilt_joint"/>
	<arg name="initialization_delay" default="0.0"/>
	<arg name="send_constantly" default="true"/>

	<param name="hostname" value="$(arg ip_address)"/>
-	<param name="username" value="$(arg username)" />
-	<param name="password" value="$(arg password)" />
	<param name="node_name" value="$(arg node_name)"/>
-	<param name="camera_model" value="$(arg camera_model)"/>
	<param name="enable_auth" value="$(arg enable_auth)"/>
	<param name="pan_joint" value="$(arg pan_joint)"/>
	<param name="tilt_joint" value="$(arg tilt_joint)"/>
	<param name="initialization_delay" value="$(arg initialization_delay)"/>
	<param name="send_constantly" value="$(arg send_constantly)"/>

	<rosparam file="$(arg ptz_config_file)" command="load" if="$(arg ptz_enabled)"/>
	<node pkg="axis_camera" type="axis_ptz_node.py" name="$(arg node_name)_ptz" output="screen"/>
</launch>
